File: 13-Algorithms_sorting/bubble_sort.rb
# frozen_string_literal: true

def bubble_sort(arr)
  return arr if arr.length < 2

  loop do
    swapped = false

    (0...arr.length - 1).each do |b|
      next unless arr[b] > arr[b + 1]

      # Swap numbers
      arr[b], arr[b + 1] = arr[b + 1], arr[b]
      swapped = true
    end

    break unless swapped
  end

  arr
end

numbers = [99, 44, 6, 2, 1, 5, 63, 87, 283, 4, 0]
p bubble_sort(numbers)

File: 13-Algorithms_sorting/bubble_sort.js
const bubbleSort = arr => {
  const length = arr.length;

  for (let i = 0; i < length; i++) {
    for (let j = 0; j < length; j++) {
      if (arr[j] > arr[j + 1]) {
        // Swap numbers
        const temp = arr[j];
        arr[j] = arr[j + 1];
        arr[j + 1] = temp;
      }
    }
  }

  return arr;
};

const numbers = [99, 44, 6, 2, 1, 5, 63, 87, 283, 4, 0];
console.log(bubbleSort(numbers));
